docs/data_types/tensor/tensor.md

```diff
@@ -0,0 +1,125 @@
+# Tensor
+
+
+
+DocArray support several tensor types that can be used inside BaseDoc. 
+
+The main ones are:
+
+- [`NdArray`][docarray.typing.tensor.NdArray] for numpy tensors
+- [`TorchTensor`][docarray.typing.tensor.TorchTensor] for Pytorch tensors
+- [`TensorFlowTensor`][docarray.typing.tensor.TensorFlowTensor] for tensorflow tensors
+
+The three of them wrap from their respective framework tensor type. 
+
+!!! note
+    [`NdArray`][docarray.typing.tensor.NdArray] and [`TorchTensor`][docarray.typing.tensor.TorchTensor] are a subclass of their native tensor type. This means that they can be used natively inside their framework.
+
+!!! note
+    [`TensorFlowTensor`][docarray.typing.tensor.TensorFlowTensor] stores the pure `tf.Tensor` object inside the `tensor` attribute. This is due to a limitation on the TensorFlow framework that does not allow to subclass the `tf.Tensor` object.
+
+DocArray also supports [`AnyTensor`][docarray.typing.tensor.AnyTensor] which is the Union of the three previous tensor types. 
+This is a generic placeholder to specify that it can work with any tensor type (numpy, torch, tensorflow).
+
+## Tensor type validation
+
+You don't need to directly instantiate the  [`NdArray`][docarray.typing.tensor.NdArray] , [`TorchTensor`][docarray.typing.tensor.TorchTensor], or [`TensorFlowTensor`][docarray.typing.tensor.TensorFlowTensor] by yourself.
+
+Instead, you should use them as type hints on [`BaseDoc`][docarray.base_doc.doc.BaseDoc] fields, where they perform data validation.
+During this process, [`BaseDoc`][docarray.base_doc.doc.BaseDoc] will cast the native tensor type into the respective DocArray tensor type.
+
+Let's take an example:
+
+```python
+from docarray import BaseDoc
+from docarray.typing import NdArray
+
+import numpy as np
+
+
+class MyDoc(BaseDoc):
+    tensor: NdArray
+
+
+doc = MyDoc(tensor=np.zeros(100))
+
+assert isinstance(doc.tensor, NdArray)  # True
+``` 
+Here you see that the `doc.tensor` is an `NdArray`. 
+
+```python
+assert isinstance(doc.tensor, np.ndarray)  # True as well
+``` 
+
+But since it inherits from `np.ndarray` you can also use it as a normal numpy array. The same holds for pytorch and `TorchTensor`.
```

Comment:
```suggestion
But since it inherits from `np.ndarray`, you can also use it as a normal NumPy array. The same holds for PyTorch and `TorchTensor`.
```